satellite/satellitedb/usercredits.go

```diff
@@ -70,37 +68,58 @@ func (c *usercredits) Create(ctx context.Context, userCredit console.UserCredit)
 		referrerID = userCredit.ReferredBy[:]
 	}
 
-	var result sql.Result
-	var err error
+	var shouldCreate bool
```

Comment:
this seems like it would be simple to do:
```
shouldCreate := false
if !rewards.Partner {
  shouldCreate = userCredit.OfferInfo.Status.IsDefault()
}
```

Reply:
I was thinking that since the logic for giving credit is based on different reward type, should we just create a separate method to handle. I imagine logic may change in the future for the partner once we don't have to bypass the token logic